internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -1189,6 +1350,10 @@ func (r *AgentServiceConfigReconciler) newAssistedServiceDeployment(ctx context.
 		newSecretEnvVar("EC_PRIVATE_KEY_PEM", "ec-private-key.pem", agentLocalAuthSecretName),
 	}
 
+	if r.exposeIPXERoute(instance) {
+		envSecrets = append(envSecrets, corev1.EnvVar{Name: "HTTP_LISTEN_PORT", Value: serviceHTTPPort.String()})
```

Comment:
This is strange, this isn't a secret so I think I'd expect this to be in the configmap rather than on the deployment.

Reply:
`envSecrets` is just an unfortunate name - we inject plain `env` using this object. Usually its references from the secret (like DB), but nothing is preventing us from injecting plain envs there

Reply:
Reworked this to inject the value in the configmap (replacing now defunct `SERVE_HTTPS`)